pkg/kvstore/consul.go

```diff
@@ -361,6 +354,59 @@ func (c *ConsulClient) GASNewL3n4AddrID(basePath string, baseID uint32, lAddrID
 	}
 }
 
+// StartWatch starts watching for changes in a prefix
+func (c *ConsulClient) StartWatch(w *Watcher) {
+	go func(w *Watcher) {
+		nextIndex := uint64(0)
+
+		// block Get() calls for 5 seconds maximum
+		qo := consulAPI.QueryOptions{
+			WaitTime: time.Duration(5) * time.Second,
+		}
+
+		for {
+			// if all goes well we don't sleep between watch cycles
+			sleepTime := time.Duration(1) * time.Millisecond
+
+			qo.WaitIndex = nextIndex
+			res, q, err := c.KV().Get(w.prefix, &qo)
```

Comment:
This seems incorrect. I suppose this should be a call to List, not Get, in order to get all the keys with that prefix?!